plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerStatsCollector.java

```diff
@@ -0,0 +1,440 @@
+
+            ConcurrentHashMap<String, ColumnStatsBuilder> columnStats = new ConcurrentHashMap<>();
+            statisticsDao.getColumnStatRanges(tableObjectId).forEach(columnStatsRange -> {
+                JdbcColumnHandle column = columns.get(columnStatsRange.columnName);
+                if (column != null) {
+                    log.info(columnStatsRange.toString());
+                    ColumnStatsBuilder builder = columnStats.computeIfAbsent(columnStatsRange.columnName, ignored ->
+                            new ColumnStatsBuilder(column, rowCount));
+                    builder.add(columnStatsRange);
+                }
+                else {
+                    log.warn("No column for %s: %s", columnStatsRange, columns.keySet());
+                }
+            });
+
+            columnStats.values().forEach(builder ->
+                    tableStatistics.setColumnStatistics(builder.getColumn(), builder.build()));
+
+            return tableStatistics.build();
+        }
+    }
+
+    private record NewStatisticsDao(Handle handle)
+    {
+        public NewStatisticsDao
+        {
+            requireNonNull(handle, "handle is null");
+        }
+
+        private static final String TABLE_LEVEL_QUERY = """
+                SELECT object_id, sum(rows) as row_count
+                FROM sys.partitions
+                WHERE object_id = object_id(:table)
+                  AND index_id IN (0, 1) -- 0 = heap, 1 = clustered index, 2 or greater = non-clustered index
+                GROUP BY object_id
+                """;
+
+        private static final String HISTOGRAM_QUERY = """
+                SELECT c.name as column_name,
+                       h.step_number,
+                       h.range_high_key,
+                       h.range_rows,
+                       h.equal_rows,
+                       h.distinct_range_rows
+                FROM (
+                    SELECT s.object_id, sc.column_id, s.name as stats_name
+                    FROM sys.stats AS s
+                    JOIN sys.stats_columns as sc
+                      ON s.object_id = sc.object_id AND s.stats_id = sc.stats_id
+                    WHERE s.object_id = :object_id
+                    GROUP BY s.object_id, sc.column_id, s.name
+                    HAVING count(*) = 1
+                ) AS ocn
+                JOIN sys.stats AS s
+                  ON ocn.object_id = s.object_id AND ocn.stats_name = s.name
+                JOIN sys.columns AS c
+                  ON ocn.object_id = c.object_id AND ocn.column_id = c.column_id
+                CROSS APPLY sys.dm_db_stats_histogram(s.object_id, s.stats_id) h
+                """;
+
+        Optional<TableLevelInfo> getTableLevelInfo(String catalog, String schema, String tableName)
+        {
+            return Optional.ofNullable(handle.createQuery(TABLE_LEVEL_QUERY)
+                    .bind("table", format("%s.%s.%s", catalog, schema, tableName))
```

Comment:
Do we need any quoting/escaping of the identifiers?

Reply:
I've just copy & pasted what already exists here as a draft. I've also been concerned with this existing code could have `null` for either catalog or schema, but again, i've simply not been addressing that yet.